arch/arm/core/aarch32/Kconfig

```diff
@@ -257,6 +257,12 @@ config ARM_STORE_EXC_RETURN
 	  This is needed when switching between threads that differ in either
 	  FPU usage or security domain.
 
+config ARM_CUSTOM_REBOOT
```

Comment:
why  is custom reboot an ARM only config? AFAIK Previously with weak symbols this would have worked for all architectures. I would change this to become a gloabl config that can be used by all platforms, not only ARM.

Reply:
Because not all architectures may need custom hooks. Some architectures will not even have a default hook. If preferred, I can create a global `CONFIG_REBOOT_NO_ARCH_DEFAULT` that can be shared, though

Reply:
note: the goal here is to remove usage of non-standard C features

Reply:
> Because not all architectures may need custom hooks. Some architectures will not even have a default hook. If preferred, I can create a global `CONFIG_REBOOT_NO_ARCH_DEFAULT` that can be shared, though

hmm, not sure what `CONFIG_REBOOT_NO_ARCH_DEFAULT`  is about, just move this config outside ARM and make it generic, this way all platforms can override architecture implementation and be happy, I do not see why this needs to be ARM only.

> note: the goal here is to remove usage of non-standard C features

you are referring to _weak symbols? I am fine with that, but we need to keep the implementation compatible here.

Reply:
> > Because not all architectures may need custom hooks. Some architectures will not even have a default hook. If preferred, I can create a global `CONFIG_REBOOT_NO_ARCH_DEFAULT` that can be shared, though
> 
> hmm, not sure what `CONFIG_REBOOT_NO_ARCH_DEFAULT` is about, just move this config outside ARM and make it generic, this way all platforms can override architecture implementation and be happy, I do not see why this needs to be ARM only.

will add a generic symbol

> 
> > note: the goal here is to remove usage of non-standard C features
> 
> you are referring to _weak symbols? I am fine with that, but we need to keep the implementation compatible here.

keeping things compatible and using non-standard C features is incompatible :-)